### PR TITLE
Lenovo cnos vrf

### PR DESCRIPTION
**Pull request by amuraleedhar**
##### SUMMARY
Introducing new module cnos_vrf to facilitate VRF configuration on cnos devices. 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/network/cnos/cnos_vrf.py
test/integration/targets/cnos_vrf/aliases
test/integration/targets/cnos_vrf/cnos_vrf_sample_hosts
test/integration/targets/cnos_vrf/defaults/main.yaml
test/integration/targets/cnos_vrf/tasks/cli.yaml
test/integration/targets/cnos_vrf/tasks/main.yaml
test/integration/targets/cnos_vrf/tests/cli/basic.yaml
test/units/modules/network/cnos/fixtures/cnos_vrf_config.cfg
test/units/modules/network/cnos/test_cnos_vrf.py

test/integration/targets/cnos_command/cnos_command_sample_hosts
test/integration/targets/cnos_config/cnos_config_sample_hosts
test/integration/targets/cnos_facts/cnos_facts_sample_hosts
test/integration/targets/cnos_lldp/cnos_lldp_sample_hosts
test/integration/targets/cnos_logging/cnos_logging_sample_hosts
test/integration/targets/cnos_showrun/cnos_showrun_sample_hosts
test/integration/targets/cnos_user/cnos_user_sample_hosts
test/integration/targets/cnos_vlan/cnos_vlan_sample_hosts

##### ANSIBLE VERSION
ansible 2.8.0.dev0 (devel e2b9c36080) last updated 2018/10/12 13:58:41 (GMT +550)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ansible/sheru/ansible/lib/ansible
  executable location = /home/ansible/sheru/ansible/bin/ansible
  python version = 2.7.6 (default, Nov 23 2017, 15:49:48) [GCC 4.8.4]

##### ADDITIONAL INFORMATION
Fixing few documentation issues also along with this PR. The code has been tested against CNOS v10.9.1

### CODE REVIEW

File: lib/ansible/modules/network/cnos/cnos_vrf.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+        should be configured in the VRF. Interfaces must be routed
+        interfaces in order to be placed into a VRF. The name of interface
+        should be in expanded format and not abbreviated.
+  associated_interfaces:
+    description:
+      - This is a intent option and checks the operational state of the for
+        given vrf C(name) for associated interfaces. If the value in the
+        C(associated_interfaces) does not match with the operational state of
+        vrf interfaces on device it will result in failure.
+    version_added: "2.5"
```
Comment (justjais) [inline lib/ansible/modules/network/cnos/cnos_vrf.py]: @amuraleedhar can you remove the `version_added` field as module field is added in `2.8`

File: lib/ansible/modules/network/cnos/cnos_vrf.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+            obj_in_want = search_obj_in_list(h['name'], want)
+            if not obj_in_want:
+                commands.append('no vrf context %s' % h['name'])
+
+    return commands
+
+
+def map_config_to_obj(module):
+    objs = []
+    output = run_commands(module, {'command': 'show vrf'})
```
Comment (justjais) [inline lib/ansible/modules/network/cnos/cnos_vrf.py:201]: please check if output is not `None` before operating on the same.

**Review by justjais — COMMENTED** [review]
@amuraleedhar have commented few cosmetic changes, please update the PR fixing the review comments

File: lib/ansible/modules/network/cnos/cnos_vrf.py (hunk 1 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_vrf.py]: Removed.

File: lib/ansible/modules/network/cnos/cnos_vrf.py (hunk 2 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_vrf.py:201]: Done

File: lib/ansible/modules/network/cnos/cnos_vrf.py [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+    if output is None:
+        return objs
```
Comment (justjais) [inline lib/ansible/modules/network/cnos/cnos_vrf.py]: @amuraleedhar wanted to check if return empty list `objs` is correct expectation here, coz fn. `map_obj_to_commands` and `check_declarative_intent_params` will be operating over empty list which makes no sense, or can we just skip play execution via `module.fail_json` with relevant msg.

Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_vrf.py]: Ok. Will do that way

**Review by justjais — APPROVED** [review]

**Pull request merged**